src/SampleExtension/Extension.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Dynamo.Extensions;
+using Dynamo.Graph.Nodes;
+using Dynamo.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace SampleExtension
+{
+    public class Extension : IExtension
+    {
+        public List<NodeModel> nodes = new List<NodeModel>();
+        public bool readyCalled = false;
+
+        public string Name
+        {
+            get
+            {
+                return "testExtension";
+            }
+        }
+
+        public string UniqueId
+        {
+            get
+            {
+                return Guid.NewGuid().ToString();
+            }
+        }
+
+        public void Dispose()
+        {
+           
+        }
+
+        public void Ready(ReadyParams sp)
+        {
+            sp.CurrentWorkspaceModel.NodeAdded += CurrentWorkspaceModel_NodeAdded;
+            this.readyCalled = true;
```

Comment:
A comment here about `ReadyParams` would also be helpful.  I think it will be useful for 3rd party devs understanding the differences between `ReadyParams`, `ViewStartupParams`, `ViewLoadedParams`, etc